tests/unit/networking.tcl

```diff
@@ -34,3 +34,33 @@ test {CONFIG SET bind address} {
         $rd close
     }
 }
+
+start_server {config "minimal.conf"} {
+    test {Default bind address configuration handling} {
+        # Default is explicit and sane
+        assert_equal "* -::*" [lindex [r CONFIG GET bind] 1]
+
+        # CONFIG REWRITE acknowledges this as a default
+        r CONFIG REWRITE
+        assert_equal 0 [count_message_lines [srv 0 config_file] bind]
+
+        # Removing the bind address works
+        r CONFIG SET bind ""
+        assert_equal "" [lindex [r CONFIG GET bind] 1]
+
+        # No additional clients can connect
+        catch {redis_client} e
+        assert_match {*connection refused*} $e
+
+        # CONFIG REWRITE handles empty bindaddr
+        r CONFIG REWRITE
+        assert_equal 1 [count_message_lines [srv 0 config_file] bind]
+
+        # Make sure we're able to restart
+        restart_server 0 0 0 0
+        exec src/redis-cli -s [srv 0 unixsocket] config set bind *
```

Comment:
if no restart redis would mess up and bind to something anyway, this test won't notice it.
i think you need to copy the CONFIG GET and "connection refused" checks after the restart.